src/main/java/org/cyclonedx/gradle/utils/ExternalResourcesUtil.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of CycloneDX Gradle Plugin.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.gradle.utils;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import org.cyclonedx.model.ExternalReference;
+
+public class ExternalResourcesUtil {
+
+    private ExternalResourcesUtil() {}
+
+    public static List<ExternalReference> createExternalReferences() {
+        final List<ExternalReference> references = new ArrayList<>();
+        addGitReference(references);
+        return references;
+    }
+
+    public static void addGitReference(final List<ExternalReference> references) {
+        try {
+            Process process = Runtime.getRuntime().exec("git remote get-url origin");
```

Comment:
Will it work for any OS?

Reply:
Yes, should work on any OS, but I will give it a shot on my windows machine as well. I wrote some test cases to reproduce these cases